docs/rfc-microloans.md

```diff
@@ -0,0 +1,477 @@
+
+**Inputs:**
+- User adds: 0.0001 BTC ($10)
+- Current BTC price: $100,000
+
+**Calculations:**
+- New user collateral = 0.0002889375 BTC + 0.0001 BTC = 0.0003889375 BTC
+- New user CR = (0.0003889375 * 100,000) / 25.125) * 100% = 154.8%
+
+**Expected State After Adding Collateral:**
+```
+Main Trove:
+- Collateral: 0.0603889375 BTC
+- Debt: 2025 MUSD
+- CR: 298.2%
+
+User MicroTrove:
+- Collateral: 0.0003889375 BTC
+- Debt: 25.125 MUSD
+- CR: 154.8%
+```
+
+#### Test Vector 4: Increasing Debt on Microloan
+
+**Inputs:**
+- User wants to borrow additional: 5 MUSD
+- BTC price: $100,000
+- Origination fee: 0.5%
+- Starting from state after adding collateral (Test Vector 3)
+
+**Calculations:**
+- Additional origination fee = 5 MUSD * 0.5% = 0.025 MUSD
+- Additional total debt = 5 MUSD + 0.025 MUSD = 5.025 MUSD
+- New total user debt = 25.125 MUSD + 5.025 MUSD = 30.15 MUSD
+- New user CR = (0.0003889375 BTC * $100,000) / 30.15 MUSD * 100% = 129.0%
+
+**Expected State After Increasing Debt:**
+```
+Main Trove:
+- Collateral: 0.0603889375 BTC ($6,038.89)
+- Debt: 2,030 MUSD (2025 + 5, only the borrowed amount)
+- CR: 297.5%
+
+User MicroTrove:
+- Collateral: 0.0003889375 BTC ($38.89)
+- Debt: 30.15 MUSD (includes origination fees)
+- CR: 129.0%
+```
+
+#### Test Vector 5: Interest Accrual
+
+**Inputs:**
+- Time elapsed: 1 year
+- User debt: 30.15 MUSD (from Test Vector 4)
+- Microloan interest rate: 5% APR (simple interest, no compounding)
+- Main trove interest rate: 1% APR (simple interest, no compounding)
+- BTC price: $100,000
+
+**Calculations:**
+- Microloan interest accrued = 30.15 MUSD * 5% = 1.5075 MUSD
+- New total user debt = 30.15 MUSD + 1.5075 MUSD = 31.6575 MUSD
+- Main trove interest accrued = 2,030 MUSD * 1% = 20.30 MUSD
+- New main trove debt = 2,030 MUSD + 20.30 MUSD = 2,050.30 MUSD
+- New user CR = (0.0003889375 BTC * $100,000) / 31.6575 MUSD * 100% = 122.8%
+- New main trove CR = (0.0603889375 BTC * $100,000) / 2,050.30 MUSD * 100% = 294.5%
+
+**Expected State After Interest Accrual:**
+```
+Main Trove:
+- Collateral: 0.0603889375 BTC ($6,038.89)
+- Debt: 2,050.30 MUSD (2030 + 20.30)
+- CR: 294.5%
+
+User MicroTrove:
+- Collateral: 0.0003889375 BTC ($38.89)
+- Debt: 31.6575 MUSD
+- CR: 122.8%
+```
+
+#### Test Vector 6: Liquidation Scenario
+
+**Inputs:**
+- Starting from state after interest accrual (Test Vector 5)
+- BTC price drops to: $93,604
+- User debt: 31.6575 MUSD
+- User collateral: 0.0003889375 BTC
+- Microloans minimum CR: 115%
+
+**Calculations:**
+- New collateral value = 0.0003889375 BTC * $93,604 = $36.40
+- New user CR = ($36.40 / 31.6575 MUSD) * 100% = 114.98%
+- Since 114.98% < 115% (microloans minimum CR), loan is eligible for liquidation
+
+**Liquidation Execution:**
+- Liquidator provides: 31.6575 MUSD
+- Liquidator receives: 0.0003889375 BTC ($36.40)
+- Liquidator profit = $36.40 - $31.6575 = $4.74
+- Main trove debt decreases by: 30 MUSD (only the borrowed amount, not the fees)
```

Comment:
What happens with 1.5075 MUSD fee accrued for the MicroTrove? We can put it aside for withdrawal later or we can reduce the main trove's debt by more than 30 MUSD. Second feels easier to implement but I am not sure if it does not create some accounting issues with the main trove later.